ci-operator/templates/openshift/openshift-azure/build-base-image.yaml

```diff
@@ -0,0 +1,240 @@
+    activeDeadlineSeconds: 10800
+    terminationGracePeriodSeconds: 600
+    volumes:
+    - name: artifacts
+      emptyDir: {}
+    - name: shared-tmp
+      emptyDir: {}
+    - name: shared-data
+      emptyDir: {}
+    - name: cluster-secrets-azure
+      secret:
+        secretName: e2e-azure-secret
+    - name: openshift-azure
+      emptyDir: {}
+
+    containers:
+    - name: buildimage
+      image: ${IMAGE_ANSIBLE}
+      resources:
+        requests:
+          cpu: 1
+          memory: 300Mi
+        limits:
+          cpu: 3
+          memory: 4Gi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /usr/share/ansible/openshift-ansible/inventory/dynamic/injected
+      env:
+      - name: HOME
+        value: /usr/share/ansible/openshift-ansible
+      - name: AZURE_VM_SIZE
+        value: "${AZURE_VM_SIZE}"
+      - name: JOB_NAME_SAFE
+        value: "${JOB_NAME_SAFE}"
+      - name: JOB_NAME_HASH
+        value: "${JOB_NAME_HASH}"
+      - name: OPENSHIFT_RELEASE
+        value: "${OPENSHIFT_RELEASE}"
+      - name: TYPE
+        value: azure
+      - name: DEPLOY_OS
+        value: "${DEPLOY_OS}"
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      command:
+      - /usr/local/bin/entrypoint-provider
+      args:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        # error handling and sync code
+        set -euo pipefail
+
+        # We need to persist this variable so that we can clean it up when we exit
+        echo "RESOURCE_GROUP=${INSTANCE_PREFIX}-$(TZ=Etc/UTC date +%Y%m%d%H%M)" >>/tmp/shared/environment
+        echo "IMAGE_NAME=${DEPLOY_OS}-$OPENSHIFT_RELEASE-$(TZ=Etc/UTC date +%Y%m%d%H%M)" >>/tmp/shared/environment
+
+        trap 'rc=$?; if [[ $rc -ne 0 ]]; then
+          touch /tmp/shared/exit;
+        fi;
+        az group delete -n ${RESOURCE_GROUP} -y --no-wait
+        exit $rc' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        # Get resource group and image name
+        source /tmp/shared/environment
+        echo "RESOURCE_GROUP=${RESOURCE_GROUP}"
+        echo "IMAGE_NAME=${IMAGE_NAME}"
+
+        source /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/secret
+        az login --service-principal -u ${AZURE_CLIENT_ID} -p ${AZURE_CLIENT_SECRET} --tenant ${AZURE_TENANT_ID} &>/dev/null
+        # Unfortunately az login is not enough for the node build playbook, we also
+        # have to source the following variables: AZURE_TENANT and AZURE_SECRET
+        # instead of AZURE_TENANT_ID and AZURE_CLIENT_SECRET which are used elsewhere.
+        cp /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/secret /tmp/shared/credentials
+        sed -i -e "s#TENANT_ID#TENANT#" /tmp/shared/credentials
+        sed -i -e "s#CLIENT_SECRET#SECRET#" /tmp/shared/credentials
+        source /tmp/shared/credentials
+
+        ansible-playbook \
```

Comment:
done

Reply:
I had to add `-e openshift_azure_vm_ssh_public_key=` to the `ansible-playbook` command since an ssh key is required for ansible to get unto the vm and perform actions like updating it.

Reply:
Also note that you will need to pass an env variable for the certs
```yaml
-e @/usr/share/ansible/openshift-ansible/inventory/dynamic/injected/certs.yaml \
```

Reply:
done